napari/layers/shapes/_shapes_utils.py

```diff
@@ -78,10 +78,42 @@ def _is_convex(poly: npt.NDArray) -> bool:
     thrd = poly[2:]
     orn_set = np.unique(orientation(fst.T, snd.T, thrd.T))
     if orn_set.size != 1:
-        return False
-    return (orn_set[0] == orientation(poly[-2], poly[-1], poly[0])) and (
-        orn_set[0] == orientation(poly[-1], poly[0], poly[1])
-    )
+        return False, 0
+    return (
+        (orn_set[0] == orientation(poly[-2], poly[-1], poly[0]))
+        and (orn_set[0] == orientation(poly[-1], poly[0], poly[1]))
+    ), int(orn_set[0])
+
+
+def _is_simple(poly: npt.NDArray, orientation_: int) -> bool:
```

Comment:
why is there a trailing `_` in `orientation_`? Can you add a docstring that explains what "simple" means, and how this function computes the answer?

Reply:
There is the `orientation` function in global scope so I need to use a different name.

Reply:
Yes. But it generates warnings in multiple linting tools.

Reply:
@Czaki Let's try to get the docstring in for this since your latest changes to CI has this passing the pretest builds. Then we can revisit @jni's comment.

Reply:
There is added docstring already

Reply:
> Yes. But it generates warnings in multiple linting tools.

imho that is a problem with the linting tools, and it is not a reason to make the code more confusing or difficult to read. I would strongly suggest removing the `_`. However, I won't hold up the PR for it.

Reply:
I thought the same way in the past until I wasted multiple hours on understanding why code stopped working after some refactor that moved code between functions.